opal/mca/memory/patcher/memory_patcher_component.c

```diff
@@ -90,7 +90,7 @@ opal_memory_patcher_component_t mca_memory_patcher_component = {
        it out) */
 };
 
-#if OPAL_MEMORY_PATCHER_HAVE___SYSCALL_PROTO && OPAL_MEMORY_PATCHER_HAVE___SYSCALL
+#if HAVE_DECL___SYSCALL && defined(HAVE___SYSCALL)
```

Comment:
nope
if you give it a closer look, you will notice the test is
```c
#if foo && defined(bar)
```

two distinct macros with two distinct usages for two distinct purposes, courtesy of autotools

fwiw

```
AC_CHECK_FUNCS([foo])
```
will either generate
```#undef HAVE_foo``` or ```#define HAVE_foo 1```
whereas
```
AC_CHECK_DECLS([foo])
```
will either generate
```#define HAVE_DECL_foo 0``` or ```#define HAVE_DECL_foo 1```